--- a/bigbluebutton-web/grails-app/conf/spring/resources.groovy
+++ b/bigbluebutton-web/grails-app/conf/spring/resources.groovy
@@ -5,7 +5,7 @@ import org.slf4j.LoggerFactory
 Logger logger = LoggerFactory.getLogger("org.bigbluebutton.web.services.turn.StunTurnService")
 
 beans = {
-  def turnConfigFilePath = "/etc/bigbluebutton/turn-stun-servers.xml"
+  def turnConfigFilePath = "@TURN_STUN_SERVERS@"
   def turnConfigFile = new File(turnConfigFilePath)
   if (turnConfigFile.canRead()) {
     logger.info("Reading stun/turn server config from overlay config file " + turnConfigFilePath)
