diff --git a/node/hooks/express/specialpages.js b/node/hooks/express/specialpages.js
index 035074e0..55dc89be 100644
--- a/node/hooks/express/specialpages.js
+++ b/node/hooks/express/specialpages.js
@@ -26,13 +26,13 @@ exports.expressCreateServer = function (hook_name, args, cb) {
   //serve robots.txt
   args.app.get('/robots.txt', function(req, res)
   {
-    var filePath = path.join(settings.root, "src", "static", "skins", settings.skinName, "robots.txt");
+    var filePath = path.join(settings.root, "static", "skins", settings.skinName, "robots.txt");
     res.sendFile(filePath, function(err)
     {
       //there is no custom favicon, send the default robots.txt which dissallows all
       if(err)
       {
-        filePath = path.join(settings.root, "src", "static", "robots.txt");
+        filePath = path.join(settings.root, "static", "robots.txt");
         res.sendFile(filePath);
       }
     });
@@ -96,14 +96,14 @@ exports.expressCreateServer = function (hook_name, args, cb) {
   //serve favicon.ico from all path levels except as a pad name
   args.app.get( /\/favicon.ico$/, function(req, res)
   {
-    var filePath = path.join(settings.root, "src", "static", "skins", settings.skinName, "favicon.ico");
+    var filePath = path.join(settings.root, "static", "skins", settings.skinName, "favicon.ico");
 
     res.sendFile(filePath, function(err)
     {
       //there is no custom favicon, send the default favicon
       if(err)
       {
-        filePath = path.join(settings.root, "src", "static", "favicon.ico");
+        filePath = path.join(settings.root, "static", "favicon.ico");
         res.sendFile(filePath);
       }
     });
diff --git a/node/utils/Settings.js b/node/utils/Settings.js
index 28d165fe..69cbfd5c 100644
--- a/node/utils/Settings.js
+++ b/node/utils/Settings.js
@@ -600,7 +600,7 @@ exports.reloadSettings = function reloadSettings() {
 
   // checks if skinName has an acceptable value, otherwise falls back to "no-skin"
   if (exports.skinName) {
-    const skinBasePath = path.join(exports.root, "src", "static", "skins");
+    const skinBasePath = path.join(exports.root, "static", "skins");
     const countPieces = exports.skinName.split(path.sep).length;
 
     if (countPieces != 1) {
